osf/external/spam/tasks.py

```diff
@@ -80,3 +90,67 @@ def _extract_domains(content):
         if domain not in extracted_domains:
             extracted_domains.add(domain)
             yield domain
+
+
+@run_postcommit(once_per_request=False, celery=True)
+@celery_app.task(ignore_results=False, max_retries=5, default_retry_delay=60)
+def check_resource_with_spam_services(guid, content, author, author_email, request_kwargs):
+    """
+    Return statements used only for debugging and recording keeping
+    """
+    any_is_spam = False
+    from osf.models import Guid, OSFUser
+    guid = Guid.load(guid)
+    resource = guid.referent
+
+    kwargs = dict(
+        user_ip=request_kwargs.get('remote_addr'),
+        user_agent=request_kwargs.get('user_agent'),
+        referrer=request_kwargs.get('referer'),
+        comment_content=content,
+        comment_author=author,
+        comment_author_email=author_email,
+        content=content,
+    )
+
+    spam_clients = []
+    if settings.AKISMET_ENABLED:
+        spam_clients.append(AkismetClient())
+    if settings.OOPSPAM_ENABLED:
+        spam_clients.append(OOPSpamClient())
+
+    for client in spam_clients:
+        is_spam, details = client.check_content(**kwargs)
+        if is_spam:
+            any_is_spam = True
+            if not resource.spam_data.get('who_flagged'):
+                resource.spam_data['who_flagged'] = client.NAME
+            elif resource.spam_data['who_flagged'] != client.NAME:
+                resource.spam_data['who_flagged'] = 'both'
+
+            if client.NAME == 'akismet':
+                resource.spam_pro_tip = details
+            if client.NAME == 'oopspam':
+                resource.spam_data['oopspam_data'] = details
+
+    if any_is_spam:
+        resource.spam_data['headers'] = {
+            'Remote-Addr': request_kwargs.get('remote_addr'),
+            'User-Agent': request_kwargs.get('user_agent'),
+            'Referer': request_kwargs.get('referer'),
+        }
+        resource.spam_data['content'] = content
+        resource.spam_data['author'] = author
+        resource.spam_data['author_email'] = author_email
+        resource.flag_spam()
+
+        if hasattr(resource, '_check_spam_user'):
+            user = OSFUser.objects.get(username=author_email)
+            resource._check_spam_user(user)
+        elif hasattr(resource, 'suspend_spam_user'):
+            user = OSFUser.objects.get(username=author_email)
+            resource.suspend_spam_user(user)
```

Comment:
I don't think that there should be a case where something has one of these functions but not the other. Honestly, `_check_spam_user` should be the "public" interface function and `suspend_spam_user` should be the "protected" function, since the whole purpose of `_check_spam_user` is to determine whether to call `suspend_spam_user`